src/app/games/siam/SiamMove.ts

```diff
@@ -1,75 +1,65 @@
-import { Orthogonal } from 'src/app/jscaip/Direction';
+import { Orthogonal, OrthogonalNumberEncoder } from 'src/app/jscaip/Direction';
 import { MGPOptional } from 'src/app/utils/MGPOptional';
 import { NumberEncoder } from 'src/app/utils/Encoder';
+import { MGPFallible } from 'src/app/utils/MGPFallible';
+import { MGPOptionalNumberEncoder } from 'src/app/utils/MGPOptionalEncoder';
 
 export class SiamMove extends MoveCoord {
-    public static encoder: NumberEncoder<SiamMove> = new class extends NumberEncoder<SiamMove> {
-        public maxValue(): number {
-            return 245 * 8 + 49 * 8 + 7 * 7 + 7;
-        }
-        public encodeNumber(move: SiamMove): number {
-            const y: number = move.coord.y + 1; // 0 to 6
-            const x: number = move.coord.x + 1; // 0 to 6
-            const moveDirection: number =
-                move.moveDirection.isAbsent() ? 4 : move.moveDirection.get().toInt(); // 0 to 4
-            const landingOrientation: number = move.landingOrientation.toInt();
-            return (245 * landingOrientation) + (49 * moveDirection) + (7 * x) + y;
-        }
-        public decodeNumber(encodedMove: number): SiamMove {
-            const y: number = encodedMove%7;
-            encodedMove -= y;
-            encodedMove/= 7;
-            const x: number = encodedMove%7;
-            encodedMove -= x;
-            encodedMove/= 7;
-            const moveDirectionInt: number = encodedMove % 5;
-            const moveDirection: MGPOptional<Orthogonal> = moveDirectionInt === 4 ?
-                MGPOptional.empty() :
-                Orthogonal.factory.fromInt(moveDirectionInt).toOptional();
-            encodedMove -= moveDirectionInt;
-            encodedMove /= 5;
-            const landingOrientation: Orthogonal = Orthogonal.factory.fromInt(encodedMove).get();
-            return new SiamMove(x - 1, y - 1, moveDirection, landingOrientation);
-        }
-    };
-    constructor(
+    public static encoder: NumberEncoder<SiamMove> = NumberEncoder.tuple([
+        NumberEncoder.numberEncoder(6), // x (from -1 to 5)
+        NumberEncoder.numberEncoder(6), // y (from -1 to 5)
+        MGPOptionalNumberEncoder<Orthogonal>(new OrthogonalNumberEncoder()), // direction
+        new OrthogonalNumberEncoder(), // orientation
+    ], (move: SiamMove): [number, number, MGPOptional<Orthogonal>, Orthogonal] => {
+        return [move.x+1, move.y+1, move.direction, move.landingOrientation];
+    }, (fields: [number, number, MGPOptional<Orthogonal>, Orthogonal]): SiamMove => {
+        return SiamMove.of(fields[0]-1, fields[1]-1, fields[2], fields[3]).get();
+    });
+
+    private constructor(
         readonly x: number,
         readonly y: number,
-        public readonly moveDirection: MGPOptional<Orthogonal>,
+        public readonly direction: MGPOptional<Orthogonal>,
         public readonly landingOrientation: Orthogonal) {
         super(x, y);
-        this.checkValidity();
     }
```

Comment:
} must be either on signature first column or signature first row ?

Reply:
trouver des lignes qui contiennent des strictement un ) et un { marche pas ?

Reply:
bah, function `foo(): X {` est une de ces lignes !

Reply:
qui contiennent strictement 1 ) et zero ( et un { je veux dire ?

Reply:
oui exact, du coup je reviens à ma question initiale: t'as une regexp pour ça sous la main ? :p

Reply:
(\n){1}[^\(\r\n]+(\)){1}([^\(\r\n]+)(\{){1}(\r|\n)

Cette ébauche en chope certains mais y a une tuile dans la couque
(\n){1} "prend moi les lignes"
[^\(\r\n]+ "qui commencent strictement par au moins un charachtère qui n'est pas un passage à la ligne ou une ouverture de parenthèse)
(\)){1} "puis qu'est suivi par une parenthèse se fermant"
([^\(\r\n]+) "puis à nouveau moultes non \r non \n non "(""
(\{){1} "puis strictement une ouverture d'accolade"
(\r|\n) "puis qui se fini par un passage à la ligne

Mais il semble malgré ça compter des choses multi-lignes. mais outre quelques faux-match ça met en avant les sales bêtes !

Reply:
Hélà! Ignoreur :D !

Reply:
ah bouger dans le ticket rename ? :D

Reply:
tu peux ticketter oui !